apps/courses/cms_wizards.py

```diff
@@ -0,0 +1,268 @@
+"""
+CMS Wizard to add a course page
+"""
+from django import forms
+from django.template.defaultfilters import slugify
+from django.utils.translation import get_language, ugettext_lazy as _
+
+from cms.api import create_page
+from cms.forms.wizards import SlugWidget
+from cms.models import Page
+from cms.wizards.wizard_base import Wizard
+from cms.wizards.wizard_pool import wizard_pool
+
+from .models import (
+    Course,
+    CoursePage,
+    CourseSubject,
+    CourseSubjectPage,
+    COURSES_PAGE_REVERSE_ID,
+    COURSE_SUBJECTS_PAGE_REVERSE_ID,
+)
+
+
+# pylint:disable=duplicate-code
+class CourseWizardForm(forms.Form):
+    """
+    This form is used by the wizard that creates a new course page.
+    A related Course model is created for each course page.
+    """
+    title = forms.CharField(
+        max_length=255,
+        required=True,
+        widget=forms.TextInput(),
+        label=_("Page title"),
+        help_text=_("Title of the page in current language"),
+    )
+    slug = forms.CharField(
+        max_length=200,  # Should be less than 255, the "max_length" of a Page's "path" field
+        required=False,
+        widget=SlugWidget(),
+        label=_("Page slug"),
+        help_text=_("Slug of the page in current language"),
+    )
+    active_session = forms.CharField(
+        max_length=200,
+        required=True,
+        widget=forms.TextInput(),
+        label=_("Course key"),
+        help_text=_("Course key of the active session"),
+    )
+
+    def clean(self):
+        """
+        Ensure that the slug field is set or derive it from the title
+        """
+        cleaned_data = super().clean()
+        # If the slug is not explicitly set, generate it from the title
+        if cleaned_data.get("title") and not cleaned_data.get("slug"):
+            cleaned_data["slug"] = slugify(cleaned_data["title"])[:200]
```

Comment:
I think you should return `cleaned_data`

Reply:
right, also changed organizations wizard this way